_episodes/06-raster-intro.md

```diff
@@ -0,0 +1,433 @@
+which is based on the popular [`rasterio`](https://rasterio.readthedocs.io/en/latest/) package for working with rasters and [`xarray`](http://xarray.pydata.org/en/stable/) for working with multi-dimensional arrays.
+
+We will also use the [`pystac`](https://github.com/stac-utils/pystac) package to load raster from the searching results from the exercise of the previous episode. This would not be necessary if you skipped it.
+
+
+> ## Introduce the Data
+>
+> In this episode, we will continue from the `search.json` file, which we saved in an exercise from the previous episode. We will use one scene in the searching results as an example, and demonstrate the data loading and visualization.
+>
+> If you have skipped the previous episode on data accessing, you can download the same raster data from the [data repository](ToDo: add link) of this lesson.
+{: .callout}
+
+## Load a Raster and View Attributes
+
+In the previous episode, we did an exercise searching Sentinel-2 images, and then saved the searching results to `search.json`. This file contains the information on where and how to access the target images from a remote repository. We can use the function `pystac.ItemCollection.from_file()` to load the searching results as an `Item` list.
+
+
+~~~
+import pystac
+items = pystac.ItemCollection.from_file("search.json")
+~~~
+{: .language-python}
+
+In the searching results, we have 2 `Item`, corresponding to 4 Sentinel-2 scenes from March 26th and 28th in 2020. We will focus on the first scene `S2A_31UFU_20200328_0_L2A`, and load the band `B09` (central wavelength 945 nm). We can load this band using function `rioxarray.open_rasterio()`, via the Hypertext Reference `href`:
+~~~
+import rioxarray
+raster_ams_b9 = rioxarray.open_rasterio(items[0].assets["B09"].href)
+~~~
+{: .language-python}
+
+By calling the variable name in the jupyter notebook we can get a quick look at the shape and attributes of the data.
+~~~
+raster_ams_b9
+~~~
+{: .language-python}
+~~~
+<xarray.DataArray (band: 1, y: 1830, x: 1830)>
+[3348900 values with dtype=uint16]
+Coordinates:
+  * band         (band) int64 1
+  * x            (x) float64 6e+05 6.001e+05 6.002e+05 ... 7.097e+05 7.098e+05
+  * y            (y) float64 5.9e+06 5.9e+06 5.9e+06 ... 5.79e+06 5.79e+06
+    spatial_ref  int64 0
+Attributes:
+    _FillValue:    0.0
+    scale_factor:  1.0
+    add_offset:    0.0
+~~~
+{: .output}
+
+The first call to `rioxarray.open_rasterio()` opens the file from remote or local storage, and then returns a `xarray.DataArray` object. The object is stored in a variable, i.e. `raster_ams_b9`. Reading in the data with `xarray` instead of `rioxarray` also returns a `xarray.DataArray`, but the output will not contain the geospatial metadata (such as projection information). You can use a `xarray.DataArray` in calculations just like a numpy array. Calling the variable name of the `DataArray` also prints out all of its metadata information.
```

Comment:
```suggestion
The first call to `rioxarray.open_rasterio()` opens the file from remote or local storage, and then returns a `xarray.DataArray` object. The object is stored in a variable, i.e. `raster_ams_b9`. Reading in the data with `xarray` instead of `rioxarray` also returns a `xarray.DataArray`, but the output will not contain the geospatial metadata (such as projection information). You can use numpy functions or built-in Python math operators on a `xarray.DataArray` just like a numpy array. Calling the variable name of the `DataArray` also prints out all of its metadata information.
```